sympy/calculus/codomain.py

```diff
@@ -151,3 +154,78 @@ def codomain_interval(f, set_val, *sym):
     if isinstance(domain, FiniteSet):
         return FiniteSet(*[limit(func, symbol, i) if i in FiniteSet(-oo, oo)
                             else func.subs({symbol: i}) for i in domain])
+
+
+def not_empty_in(finSet_intersection, *syms):
+    """ Finds the domain of the functions in `finite_set` in which the
+    `finite_set` is not-empty
+
+    Parameters
+    ==========
+
+    finSet_intersection: The unevaluated intersection of FiniteSet containing
+                        real-valued functions with Union of Sets
+    syms: Tuple of symbols
+            Symbol for which domain is to be found
+
+    Raises
+    ======
+
+    NotImplementedError
+        The algorithms to find the non-emptiness of the given FiniteSet are
+        not yet implemented.
+    ValueError
+        The input is not valid.
+    RuntimeError
+        It is a bug, please report to the github issue tracker.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, codomain, FiniteSet, Interval, not_empty_in, sqrt, oo
+    >>> x = Symbol('x')
+    >>> not_empty_in(FiniteSet(x/2).intersect(Interval(0, 1)), x)
+    [0, 2]
+    >>> not_empty_in(FiniteSet(x, x**2).intersect(Interval(1, 2)), x)
+    [-sqrt(2), -1] U [1, 2]
+    >>> not_empty_in(FiniteSet(x**2/(x + 2)).intersect(Interval(1, oo)), x)
+    (-2, -1] U [2, oo)
+    """
+
+    # TODO: handle piecewise defined functions
+    # TODO: handle transcendental functions
+    # TODO: handle multivariate functions
+    finite_set = finSet_intersection.args[1]
+    sets = finSet_intersection.args[0]
+    if not isinstance(finite_set, FiniteSet):
+        raise ValueError('A FiniteSet must be given, not %s: %s' % (type(finite_set), finite_set))
+    if len(syms) == 1:
+        symbol = syms[0]
+    else:
+        raise NotImplementedError('more than one variables %s not handled' % (syms,))
+
+    y = Dummy('y')
+
+    def elm_domain(expr, *sym):
+        """
+        finds the domain of a single element of `finite_set` with `sets`
+        as its range and with given symbol `syms`
+        """
+        domain_union = S.EmptySet
+
+        # find the inverse of items in the finite_set
+        invert_expr = solveset_real(expr - y, symbol)
+        if isinstance(invert_expr, Intersection):
+            invert_set = invert_expr.args[1]
+        elif isinstance(invert_expr, Complement):
+            invert_set = invert_expr.args[0].args[1]
```

Comment:
I agree that it can be the case.
There are two reasons why i have not taken care of such a case
1. First i can not find an example where `solveset_real(f - y, x)` returns` Interval(-oo, oo) - FiniteSet(y)` or something similar.

2. Secondly currently we have `Interval - FiniteSet` returns `EmptySet` which has not been fixed.

Though adding this particular case would not be tough. So do you think we can take care this particular case in the `future` ?